Button edge-case refinements

aaronshekey opened the pull request:
A number of button changes here, mostly focused on little edge-case polish.

Changes on the right.

1. Focused states are important. I think we should maintain the focused border regardless of hover states.
![focus](https://cloud.githubusercontent.com/assets/1369864/6906073/0609ab68-d6f2-11e4-9a83-8ca8fad9f441.png)

2. Selected buttons should still have a hover state since you can still interact with them.
![selected-hover](https://cloud.githubusercontent.com/assets/1369864/6906000/7410124c-d6f1-11e4-9c38-a470adcc8cdb.png)

3. Primary button removes text shadow in favor of a darker color for improved readability.
![primary](https://cloud.githubusercontent.com/assets/1369864/6921904/3f9ea306-d78d-11e4-90e1-7e6852c974d6.png)

4. Green is slightly darker site-wide. Improves readability by removing some of the fluorescence.
![context](https://cloud.githubusercontent.com/assets/1369864/6921912/49a7b87e-d78d-11e4-93f0-0964dd320c68.png)

5. Primary button focus states have been adjusted slightly to account for the darker green.
![primary-focus](https://cloud.githubusercontent.com/assets/1369864/6925214/974a1bec-d7a3-11e4-8f3a-d32618a3bb56.png)

6. Focused buttons within the button groups are now the highest in the z-order as opposed to being overlapped by the button to the right.
![focus-z](https://cloud.githubusercontent.com/assets/1369864/6925202/86b7b8e8-d7a3-11e4-93e1-0532a6ca1077.png)

mdo commented:
Not stoked about the last two changes. I'm in favor of darkening the text shadow,  but not reversing the direction. The text color change on the disabled state also feels off—maybe a little muddy?

aaronshekey commented:
I can revert the shadow direction, but did so to match the light source of all the gradients throughout Primer. Do you just not like it visually or is there some rationale to having the shadow cast opposite the light source?

Darkened the disabled primary button. Is this any better or should I drop it?
![context](https://cloud.githubusercontent.com/assets/1369864/6906979/0be86366-d6f8-11e4-9c5c-11d0e1a8dcc3.png)

mdo commented:
The intent of the text-shadow is to illicit an embossed effect of sorts. The text isn't sitting on top of the button, it's inset. Still not a fan of the disabled primary button change :).

aaronshekey commented:
@mdo Okie dokie. I've reverted those last two points and updated the screengrabs. I disagree that they should be embossed, but hey, it's your huge mistake, not mine :wink:

mdo commented:
Originally I had no `text-shadow` on them, but that looked off a bit. I wouldn't be opposed to seeing the colors change so we can nuke the `text-shadow` entirely though—we don't have one on any of our other buttons I think.

aaronshekey commented:
> I wouldn't be opposed to seeing the colors change

Try these new colors out. I think it improves readability without losing the soul of the original green. It only loses a little fluorescence is all.

jasonlong commented:
Most of these are :ok_hand: by me. 

I'm not sure how I feel about darkening the primary buttons though. Most of our brand colors are pretty vibrant and this one feels pretty heavy.

![screen shot 2015-03-31 at 2 05 51 pm](https://cloud.githubusercontent.com/assets/6104/6925685/0f6418ca-d7af-11e4-9cc5-c637f9d69c3d.png)

bleikamp commented:
Definitely like most of these changes. 

I agree with @jasonlong's point, and I think my only big hang up would be the changes to the primary button color. Maybe it's something we'd use for a few days and wouldn't notice again, but it definitely feels a little out of place in the screenshots.

I would be open to adjusting our colors at some point, as I think the green, red, and orange in particular are pretty bright. But maybe that's a conversation for another PR? It seems like we could get this merged pretty quickly if we moved that to another PR.

fabianperez commented:
:-1: on the green. Everything else looks good.

aaronshekey commented:
Perhaps if we flatten the green buttons by getting rid of the top gradient highlight. I love the green as a large swatch, but when it's applied to thin type and buttons with white text, it really lacks contrast.

I think this is a really good example of that.
![all-is-well](https://cloud.githubusercontent.com/assets/1369864/6925884/484ca60e-d7a8-11e4-8943-273ed9f5e208.png)

I think the other stuff is shippable. I just worry that the focus state for buttons might get a little obnoxious in practice. I'll sit with it for a bit.

bleikamp commented:
> Perhaps if we flatten the green buttons by getting rid of the top gradient highlight

I'd like to avoid going totally flat, but I perhaps a more subtle gradient? Affordance still seems important, and one flat button seems a little off, too.

jasonlong commented:
Agreed that we should tackle any brand color and UI element tweaks separately. I think we can definitely make some improvements, but there are a lot of use cases to consider.
